src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -494,92 +481,53 @@ def _special_cases(self, text, cmd, outside):
-                continue_flag = True
-            else:  # inject into cmd
-                cmd_base = ' '.join(injected_command)
-                if all(isinstance(result, (six.text_type, six.string_types)) for result in results):
-                    for result in results:
-                        cmd_base = cmd_base.replace(SELECT_SYMBOL['query'], result, 1)
-                    self.cli_execute(cmd_base)
-                    continue_flag = True
-                elif all(isinstance(result, list) for result in results):
-                    if len(results) > 1:
-                        for res_counter, _ in enumerate(results[0]):
-                            base = cmd_base
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    else:
-                        for res_counter, _ in enumerate(results[0]):
-                            base = cmd_base
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    continue_flag = True
-
+            if len(args) == 1:  # if arguments start with query_symbol, just print query result
+                if args[0] == query_symbol:
+                    print(json.dumps(self.last.result, sort_keys=True, indent=2), file=self.output)
+                elif len(args) == 1 and args[0].startswith(query_symbol):
+                    query = args[0][symbol_len:]
+                    print(json.dumps(jmespath.search(query, input_dict), sort_keys=True, indent=2), file=self.output)
```

Comment:
I would consider restructuring these if statements.
There should only be a need for a single `print()` with a different first value for `json.dumps()`.

It's evident as this can be made clearer as it starts with `if len(args) == 1` yet `elif len(args) == 1` later one which would already be true.

Reply:
fixed